### PR TITLE
[+] DbQuery: improve to use SELECT or DELETE syntax.

### PR DESCRIPTION
**Pull request by PrestaEdit**

### CODE REVIEW

**Comment by jeromenadaud** [comment]
Hi,

Thank you for reporting, why not use directly Db::getInstance()->delete()  if you want to delete ?

Best regards.

**Comment by PrestaEdit** [comment]
Hi,

Sometimes, DBQuery let you use a more complex SQL query easy.

Best practice is to use delete() method if, as I think, you use it for a WHERE of a simple field (e.g.= WHERE id_customer = 1).

This improve of the DbQuery class to use the DELETE one can be more easy to use if you want to do a delete with more fields or if you some params like ($active ? active = 1 : '').

If you need, I can write you an example to see how I use it and see the benefits of this, :)

**Comment by jeromenadaud** [comment]
Hi,

Yes please, some example will be a good start point for PrestaShop community.
Kindly regards.

**Comment by PrestaEdit** [comment]
Hi @jeromenadaud ,

Here is two examples, a SELECT and a DELETE statement:

```<?php
/**
 * DbQuery: SELECT statement example
 */
public static function getDataValue($active = true)
{
	$query = new DbQuery();
	// $query->type('SELECT'); /* SELECT is default value, so no need to do it */

	$query->select('data_value');
	$query->from('data_table');

	if ($active)
		$query->where('active = 1');

	return Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query);
}

/**
 * DbQuery: DELETE statement example
 */
public static function processDelete($active = false)
{	
	$query = new DbQuery();

	$query->type('DELETE');
	$query->from('data_table');
	
	if ($active)
		$query->where('active = 1');

	return Db::getInstance(_PS_USE_SQL_SLAVE_)->execute($query);
}

File: classes/db/DbQuery.php [hunk 1]
```diff
@@ -216,8 +232,11 @@ public function limit($limit, $offset = 0)
 	 */
 	public function build()
 	{
-		$sql = 'SELECT '.((($this->query['select'])) ? implode(",\n", $this->query['select']) : '*')."\n";
-
+		if($this->query['type'] == 'SELECT')
```
Comment (jeromenadaud) [inline classes/db/DbQuery.php]: Norms, please add space after "if"

Comment (PrestaEdit) [inline classes/db/DbQuery.php]: OOps ! My bad, I forgot a norm ! (And you know that I like it !). I do the modification right now !

**Comment by jeromenadaud** [comment]
Hi,

Yes, why not, PrestaShop is also community contributions that help you to develop with.
Just fix norms, and I Merge.

Best regards.

**Comment by PrestaEdit** [comment]
Hi,

Norms is done ! ;-)

**Pull request merged**